lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3761,28 +3729,16 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       builder.createCopyAddr(adjLoc, source, dest, IsTake, IsInitialization);
     }
 
-    // Deallocate adjoint buffers for original indirect parameters.
-    for (auto adjBuf : reversed(origIndirectParamAdjointBuffers)) {
-      if (auto *cleanup = adjBuf.getCleanup())
-        cleanup->applyRecursively(builder, adjLoc);
-      builder.createDeallocStack(adjLoc, adjBuf);
+    // Deallocate local allocations.
+    for (auto *alloc : localAllocations) {
+      // Assert that local allocations have at least one use.
+      // Buffers should not be allocated needlessly.
+      assert(!alloc->use_empty());
+      builder.createDeallocStack(adjLoc, alloc);
```

Comment:
You need to apply cleanups.

Reply:
Done in https://github.com/apple/swift/pull/22645/commits/897fa8f73af077cecbfffe7f8f21b9a2ef0ec084.